src/modules/ejector/ejector.py

```diff
@@ -0,0 +1,198 @@
+import logging
+from functools import lru_cache, reduce
+
+from web3.types import Wei
+
+from src.constants import (
+    MAX_WITHDRAWALS_PER_PAYLOAD,
+    ETH1_ADDRESS_WITHDRAWAL_PREFIX,
+    MIN_PER_EPOCH_CHURN_LIMIT,
+    CHURN_LIMIT_QUOTIENT, FAR_FUTURE_EPOCH, MIN_VALIDATOR_WITHDRAWABILITY_DELAY,
+)
+from src.modules.ejector.data_encode import encode_data
+from src.modules.ejector.prediction import RewardsPredictionService
+from src.modules.ejector.typings import ProcessingState, ReportData
+from src.modules.submodules.consensus import ConsensusModule
+from src.modules.submodules.oracle_module import BaseModule
+from src.providers.consensus.typings import Validator
+from src.typings import BlockStamp, EpochNumber
+from src.utils.abi import named_tuple_to_dataclass
+from src.utils.validator_state import is_validator_active
+from src.web3py.extentions.lido_validators import LidoValidator, NodeOperatorIndex
+from src.web3py.typings import Web3
+
+
+logger = logging.getLogger(__name__)
+
+
+class Ejector(BaseModule, ConsensusModule):
+    """
+    1. Get withdrawals amount
+    2. Get exit prediction
+    Loop:
+     a. Get validator
+     b. Remove withdrawal amount
+     c. Increase order
+     d. Check new withdrawals epoches
+     e. If withdrawals ok - exit
+    3. Decode gotten lido validators
+    4. Send hash + send data
+    """
+    CONSENSUS_VERSION = 1
+    CONTRACT_VERSION = 1
+
+    AVG_EXPECTING_TIME_IN_SWEEP_MULTIPLIER = 0.5
+
+    def __init__(self, w3: Web3):
+        self.report_contract = w3.lido_contracts.validators_exit_bus_oracle
+        super().__init__(w3)
+
+        self.prediction_service = RewardsPredictionService(w3)
+
+    def execute_module(self, blockstamp: BlockStamp):
+        self.process_report(blockstamp)
+        report_blockstamp = self.get_blockstamp_for_report(blockstamp)
+        if report_blockstamp:
+            self.process_report(report_blockstamp)
+
+    @lru_cache(maxsize=1)
+    def build_report(self, blockstamp: BlockStamp) -> tuple:
+        validators = self.get_validators_to_eject(blockstamp)
+
+        data, data_format = encode_data(validators)
+
+        return ReportData(
```

Comment:
maybe we should coordinate and handle the sanity check limits regarding data elements max count (on-chain)